### PR TITLE
Use keccak syscall #170 

### PR DESCRIPTION
**Pull request by otselnik**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -25,7 +25,7 @@ FROM builder AS evm-loader-builder
 COPY ./evm_loader/ /opt/evm_loader/
 WORKDIR /opt/evm_loader/program
 RUN cargo build-bpf --features no-logs
-RUN cd ../cli && cargo build --release
+# RUN cd ../cli && cargo build --release
```
Comment (otselnik) [inline Dockerfile]: neon-cli have conflicts with  solana-sdk 1.6.9 needs to be updated

File: evm_loader/program/src/executor_state.rs [hunk 2]
```diff
@@ -517,7 +517,7 @@ impl<B: Backend> Backend for ExecutorState<B> {
 
     fn code_hash(&self, address: H160) -> H256 {
         self.substate.known_code(address)
-            .map(|code| Keccak256::digest(&code))
+            .map(|code| keccak256_digest(&code))
```
Comment (s-medvedev) [inline evm_loader/program/src/executor_state.rs]: Here you can use keccak256_h256 and remove the next line.

File: evm_loader/program/src/executor.rs [hunk 3]
```diff
@@ -185,20 +181,21 @@ impl<'config, B: Backend> Handler for Executor<'config, B> {
         let address =
             match scheme {
                 evm::CreateScheme::Create2 { caller, code_hash, salt } => {
-                    let mut hasher = Keccak256::new();
-                    hasher.input(&[0xff]);
-                    hasher.input(&caller[..]);
-                    hasher.input(&salt[..]);
-                    hasher.input(&code_hash[..]);
-                    H256::from_slice(hasher.result().as_slice()).into()
+                    // let mut hasher = Keccak256::new();
```
Comment (s-medvedev) [inline evm_loader/program/src/executor.rs]: Remove commented lines.

File: evm_loader/test_keccak_perf.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+from solana.transaction import AccountMeta, TransactionInstruction, Transaction
```
Comment (s-medvedev) [inline evm_loader/test_keccak_perf.py]: We will be evaluating the performance gains by running Uniswap tests, so these tests do not need to be implemented.

File: Dockerfile (hunk 1 above)
Comment (s-medvedev) [reply Dockerfile]: neon-cli used in tests and proxy so it needs to be repaired.

**Review by s-medvedev — COMMENTED** [review]
What about keccak calls inside EVM?

**Review by s-medvedev — APPROVED** [review]

**Pull request merged**